lib/SymbolGraphGen/Symbol.cpp

```diff
@@ -247,6 +257,18 @@ void Symbol::serializeDocComment(llvm::json::OStream &OS) const {
   }
 
   OS.attributeObject("docComment", [&](){
+    auto Loc = DocCommentProvidingDecl->getLoc(/*SerializedOK=*/true);
+    if (Loc.isValid()) {
+      auto FileName = DocCommentProvidingDecl->getASTContext().SourceMgr.getDisplayNameForLoc(Loc);
+      if (!FileName.empty()) {
```

Comment:
Maybe this pattern could be refactored to avoid code duplication?

Reply:
I've pushed up an attempt to split out the file name collection and serialization so that `serializeLocation` can use the same code paths as `serializeDocComment` here. It was a bit awkward to work it around `serializeLocation`, but maybe i could re-duplicate the file-name _collection_ but keep the _serialization_ refactored like this. 🤷‍♀️

Reply:
The second option is what I had in mind (just deduplicating the code that is now in `serializeFileURI`). However the new version is kinda nice, I don't think it is awkward to read. Maybe the functions shouldn't be called `getFileURI` since they don't actually return URIs but rather pathnames.

Reply:
Good call! I've amended the commit to rename the functions.